lib/request-wrapper.ts

```diff
@@ -311,6 +320,29 @@ export class RequestWrapper {
 
     return error;
   }
+
+  private gzipRequestBody(data: Object, headers: OutgoingHttpHeaders): Buffer|Object {
+    // skip compression if user has overridden the encoding header
+    if (headers['content-encoding']) {
```

Comment:
A couple things with this...
1) Do we not have to worry about the case of header names in node?   i.e. if the user set the "Content-Encoding" header (what is considered the canonicalized form of "content-encoding") would that register here?
2) Content-Encoding is one of those headers that could conceivably have more than one value.   I think the check here should be whether or not the header already has the value "gzip" rather than whether or not Content-Encoding is already set or not.

Reply:
1. I think we do have to worry about the case. I was going purely off of the design wiki but I see that the other cores use `Content-Encoding` so I'll change it.
2. That sounds good 👍